Python/ceval.c

```diff
@@ -7033,6 +7033,20 @@ PyEval_SetProfile(Py_tracefunc func, PyObject *arg)
     }
 }
 
+void
```

Comment:
It's not great that the function cannot report errors. While not returning -1 on error and pass the exception to the caller, rather than handling it with _PyErr_WriteUnraisableMsg()?

Reply:
I am mimicking what we do in `PyEval_SetProfile` and to avoid surprises I think we should keep these two as synchronized as possible.

Reply:
We cannot fix the API of old functions, but we can avoid past mistakes in newly added functions.

Reply:
If you want to ignore exceptions on purpose, please mention it in the function documentation.

Reply:
> If you want to ignore exceptions on purpose, please mention it in the function documentation.

Yeah, I want to do that because I don't want to stop setting the profile function on other threads if an exception in one fails. Unless you strongly disagree, I will document this.